Assets/MRTK/SDK/Features/UX/Scripts/Cursors/FingerCursor.cs

```diff
@@ -154,13 +154,10 @@ protected virtual bool IsNearGrabbableObject()
             var focusProvider = CoreServices.InputSystem?.FocusProvider;
             if (focusProvider != null)
             {
-                var spherePointers = focusProvider.GetPointers<SpherePointer>();
-                foreach (var spherePointer in spherePointers)
+                var spherePointer = focusProvider.GetFirstPointerWhere<SpherePointer>(p => p.Controller == Pointer.Controller);
```

Comment:
I'm not sure this change does a whole ton to the allocations here. It seems to only drop it from 144B to 112B (which yes, is _something_, but this is also in the path we're discussing the perf impacts of in another comment). I'm pretty sure the new allocs are from the predicate being passed into `GetFirstPointerWhere`. I'll investigate.

Before this change:
![image](https://user-images.githubusercontent.com/3580640/141367345-07ce37cf-e1f5-4683-b0ad-3a056cc2e6e4.png)


After: 
![image](https://user-images.githubusercontent.com/3580640/141366158-825a8b04-5fc3-46df-8c1a-77917f8a4755.png)

Reply:
Okay I reworked this to avoid the alloc! 
![image](https://user-images.githubusercontent.com/3580640/141393984-31598112-bac6-49a1-9522-96a82cd2a4a9.png)
It's a kinda-weird static pattern that I've used in the past, but it should work!